### PR TITLE
feat(#8): add upload section and fetch user`s upload request

### LINKED ISSUE #8: When Uploads section is opened, the SPA shall fetch user's upload requests.

### PR DESCRIPTION
**Pull request by G2p1**
Closes #8

### CODE REVIEW

File: src/lib/server/ApiClient.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export async function getUploadRequest() {
```
Comment (s1ckret) [inline src/lib/server/ApiClient.ts]: The function should return ready to use data or throw an error.

Please add return types for this function.
See [this section](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases) of typescript documentation

You can see API response example [here](https://github.com/S1ckret-Labs/family-archive-docs/blob/main/requirements/containers/backend/family-archive-web-server.md#get-upload-requests). It returns array of custom objects.

So define a new type `UploadRequest` which will describe the object.

Also your fetch request can fail. Please learn about error handling.

File: src/components/UploadSection.svelte [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    import {onMount} from 'svelte';
+    import { getUploadRequest } from '../lib/server/ApiCLient';
+  
+    let  files = [];
+    let uploadedFiles = [];
+
+    onMount(async()=>{
+
+        const get  = await getUploadRequest()
+        uploadedFiles = await get.json();
```
Comment (s1ckret) [inline src/components/UploadSection.svelte]: This should look something like that:
```ts
const uploadRequests: UploadRequest[] = await fetchUploadRequests()
```
Also your fetch request can fail. So learn about exception handling.

**Review by s1ckret — CHANGES_REQUESTED** [review]
Some changes needed but good overall

File: src/lib/server/ApiClient.ts (hunk 1 above)
Comment (G2p1) [reply src/lib/server/ApiClient.ts]: Ok, i learn how to handle, and maybe i need to show users error? or its just for us? becouse i only show errors in console, but now it used try/catch construction, and aded custom type for request

File: src/components/UploadSection.svelte (hunk 2 above)
Comment (G2p1) [reply src/components/UploadSection.svelte]: Thank you, make it with with type and handling exeption

File: src/components/UploadSection.svelte [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        try{
+        uploadedFiles  = await getUploadRequest();
+        if(uploadedFiles === null){
+            throw new Error;
+        }
+        }
+        catch(e){
+            console.error(e);
+            uploadedFiles = null;
+        }
```
Comment (s1ckret) [inline src/components/UploadSection.svelte:14]: 1.  Please enable auto formatting in your IDE
2. What is the point of throwing an error and then catching it ?

File: src/lib/server/ApiClient.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export async function getUploadRequest(): Promise<UploadRequest[]> {
+ 
+    try{
+        const uploadRequest: UploadRequest[] =  await fetch('https://gx9mczg2il.execute-api.us-east-1.amazonaws.com/api/v1/users/1/upload/requests')
+                                            .then(function(response) {
+                                                if(!response.ok){
+                                                    throw response.statusText;
+                                                }
+                                                return response.json();
+                                            });
```
Comment (s1ckret) [inline src/lib/server/ApiClient.ts]: Please don't mix `await/async` syntax and `Promise` syntax.
We will discuss on this later

File: src/lib/server/ApiClient.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  
+
+        return uploadRequest;
+    }
+    catch(e){
+        console.error(e);
+        return null;
+    }
```
Comment (s1ckret) [inline src/lib/server/ApiClient.ts]: What is the point of returning `null`? How is a user of the function should know what happened?
There can be a lot of different errors, such as 503 (server went down), 404 (sometimes it is ok to have this error), 403, 401 (auth related).

The point is that API client can't handle all errors by itself. E.g. if a client doesn't have an internet connection we should show a popup. When you return just `null` it is not possible.

Some errors can be handled by API client. E.g. 503 (service temporary unavailable). API client can perform retries, but if failed, the client of a function also should know about an error.

In conclusion, it is important to outline who's responsible for handling particular errors.

**Review by s1ckret — CHANGES_REQUESTED** [review]

**Review by s1ckret — APPROVED** [review]
